test/box-luatest/gh_9629_netbox_async_request_connection_gc_test.lua

```diff
@@ -0,0 +1,26 @@
+local net_box = require('net.box')
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new()
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+-- Tests that a connection with asynchronous requests does not get garbage
+-- collected.
+g.test_async_request_connection_gc = function(cg)
+    local c = net_box.connect(cg.server.net_box_uri)
+    local f = c:eval('return 777', {}, {is_async = true})
+    c = nil -- luacheck: no unused
+    collectgarbage()
+    local res, err = f:wait_result(0.1)
```

Comment:
Please increase the timeout. For example, to 1 second.
I was able to get the following error, when ran the test a ton of times in parallel:
```
[046] box-luatest/gh_9629_netbox_async_request_conne>                 [ fail ]
[046] Test failed! Output from reject file /tmp/t/rejects/box-luatest/gh_9629_netbox_async_request_connection_gc.reject:
[046] Tarantool version is 3.1.0-entrypoint-161-g4a5b793639
[046] TAP version 13
[046] 1..1
[046] # Started on Fri Mar 22 15:04:19 2024
[046] # Starting group: box-luatest.gh_9629_netbox_async_request_connection_gc
[046] not ok 1	box-luatest.gh_9629_netbox_async_request_connection_gc.test_async_request_connection_gc
[046] #   ...test/gh_9629_netbox_async_request_connection_gc_test.lua:24: expected: nil, actual: timed out
[046] #   stack traceback:
[046] #   	...test/gh_9629_netbox_async_request_connection_gc_test.lua:24: in function 'box-luatest.gh_9629_netbox_async_request_connection_gc.test_async_request_connection_gc'
[046] #   	...
[046] #   	[C]: in function 'xpcall'
[046] # Ran 1 tests in 3.030 seconds, 0 succeeded, 1 failed
[046] 
[046] [test-run server "luatest_server"] The log file /private/tmp/t/046_box-luatest/gh_9629_netbox_async_request_connection_gc_test.log has zero size
```
I was able to achieve this on my machine, meaning this error will be rather frequent in CI.

Reply:
Sorry, I haven't thought about this previously. Set the timeout to 10 seconds.